python/src/llvm.cc

```diff
@@ -56,6 +56,13 @@ std::string translateLLVMIRToASM(llvm::Module &module,
       *optPtr = true;
     }
   }
+  if (triton::tools::getBoolEnv("DISABLE_LLVM_LSR")) {
```

Comment:
Can this be generalized to disable a set of llvm optimizations, e.g, taking in a comma-separated optimization list?

Reply:
I am not sure if we have a disable flag for each optimization pass. There are list of disable-xxx flags defined in various places such as https://github.com/llvm/llvm-project/blob/main/llvm/lib/CodeGen/TargetPassConfig.cpp#L94
It may be possible to pass a list of disable-xxx flags, maybe DISABLE_LLVM_OPT_LIST=disable-xxx,disable-yyy?

Reply:
Yeah, not every pass can be disabled. For those can, maybe a general switch can be useful.

We already have a `DISABLE_LLVM_OPT` switch, maybe extend it to take a list and a value 1 that's to disable all passes?

Reply:
DISABLE_LLVM_OPT is currently for the llvm passes when lowering to llvm ir. But in this patch, I am trying to disable the passes during translate_to_asm. I am not sure if it is a good idea to use the same env variable.

Reply:
DISABLE_LLVM_OPT disables LLVM IR optimizations actually. There is no way to disable MLIR and MLIR LLVM dialect optimizations (such as CSE) for now.